### PR TITLE
Consolidate some redundant logic in code

### PR DESCRIPTION
**Pull request by tractorcow**
@mfendeksilverstripe these are the suggested changes I had in mind to follow up to #717

I switched out some of the logic in your loopy for checks / queries that were already handled in RecordLocale.

getLocaleInstances was also simplified slightly by re-using this logic.

I have another PR coming up (will take some time to progress) that adjusts the internal storage for localisations to use the localeID

### CODE REVIEW

File: src/Extension/FluentExtension.php [hunk 1]
```diff
@@ -650,55 +660,42 @@ public function onAfterWrite(): void
...
+        foreach ($this->owner->Locales() as $recordLocale) {
+            // Skip locales this record isn't localised in
+            if (!$recordLocale->IsDraft()) {
+                continue;
+            }
 
-        /** @var Locale $locale */
-        foreach ($locales as $locale) {
-            if ($locale->Locale === $currentLocale) {
+            $locale = $recordLocale->getLocaleObject();
```
Comment (mfendeksilverstripe) [inline src/Extension/FluentExtension.php:669]: [nitpick] This looks like unused variable.
```suggestion
```

File: src/Extension/FluentExtension.php [hunk 2]
```diff
@@ -707,20 +704,18 @@ static function (FluentState $state) use ($owner, $locale): ?DataObject {
...
 
-        foreach ($objectsToWrite as $localeCode => $localisedRecord) {
-            FluentState::singleton()->withState(
-                static function (FluentState $state) use ($localeCode, $localisedRecord): void {
-                    $state->setLocale($localeCode);
-                    $localisedRecord->write();
-                }
-            );
+            // Update localised record if any localised copies took place
+            if ($hasRecordsToWrite) {
```
Comment (mfendeksilverstripe) [inline src/Extension/FluentExtension.php:711]: [nitpick] Would prefer early exit approach here to reduce code nesting.

**Review by mfendeksilverstripe — APPROVED** [review]
Looks really great 👍 , the code is much cleaner and easier to understand. I left two nitpicks but these are very minor things. The tests are passing on my local but I noticed that the Travis test suite is not running.

File: src/Extension/FluentExtension.php (hunk 1 above)
Comment (tractorcow) [reply src/Extension/FluentExtension.php:669]: ah good spotting.

File: src/Extension/FluentExtension.php (hunk 2 above)
Comment (tractorcow) [reply src/Extension/FluentExtension.php:710]: you mean 

```php
if (!$hasRecordsToWrite) {
  continue
}
FluentState::singleton()->withState(
    static function (FluentState $state) use ($recordLocale, $localisedRecord): void {
        $state->setLocale($recordLocale->getLocale());
        $localisedRecord->write();
    }
);
```

Comment (mfendeksilverstripe) [reply src/Extension/FluentExtension.php:710]: That's correct ;-)

Comment (tractorcow) [reply src/Extension/FluentExtension.php:710]: I'll update when I got some time :D

**Pull request merged**